pkg/adapter/config/encryptionAdapter.go

```diff
@@ -2,29 +2,132 @@ package config
 
 import (
 	"context"
+	"errors"
+	"github.com/cloudogu/cesapp-lib/keys"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"github.com/cloudogu/k8s-dogu-operator/controllers/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type PublicKeyConfigEncryptionAdapter struct {
+	secrets   secret
+	registry  etcdRegistry
+	namespace string
 }
 
-func NewPublicKeyConfigEncryptionAdapter() *PublicKeyConfigEncryptionAdapter {
-	return &PublicKeyConfigEncryptionAdapter{}
+func NewPublicKeyConfigEncryptionAdapter(secretClient secret, registry etcdRegistry, namespace string) *PublicKeyConfigEncryptionAdapter {
+	return &PublicKeyConfigEncryptionAdapter{secrets: secretClient, registry: registry, namespace: namespace}
 }
 
 func (p PublicKeyConfigEncryptionAdapter) Encrypt(
 	ctx context.Context,
 	name common.SimpleDoguName,
 	value common.SensitiveDoguConfigValue,
 ) (common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	pubkey, err := resource.GetPublicKey(p.registry, string(name))
+	if err != nil {
+		return "", &domainservice.NotFoundError{
```

Comment:
You can use the new constructor for `NotFoundErrors`. It should be in develop by now.